### PR TITLE
Wip/don/display improvements

### LINKED ISSUE #34: Strings for display while FPGA is calibrating
See UI specification, section 4:
https://docs.google.com/document/d/1_kuz-_W_UsuEhLc8adFIaJ8OnwBQP7SH2txJTRZ7jVQ/edit

### LINKED ISSUE #35: Powering off message
See UI specification, section 4:
https://docs.google.com/document/d/1_kuz-_W_UsuEhLc8adFIaJ8OnwBQP7SH2txJTRZ7jVQ/edit

### LINKED ISSUE #37: Broken battery display
When the battery requires replacement or is missing, the battery symbol on the display should indicate this else undefined behaviour could confuse the user.
Gather requirements from UI spec.

### PR DESCRIPTION
**Pull request by donaldrobson**
Closes #37
Closes #34
Closes #35

### CODE REVIEW

**Comment by donaldrobson** [comment]
Note - broken/missing battery indicator is not defined in the UI spec, but there should be one to alleviate confusion, so I have made one.  There is currently no calling code, anyway.

**Comment by donaldrobson** [comment]
Also adds functions for formatting progress bar and/or any string onto the display, as well as macros for the strings defined in the UI spec.

File: code/drivers/fpga/display.h [hunk 1]
```diff
@@ -6,6 +6,22 @@
...
 
+#define DISPLAY_STR_HOLD_MUTE_SEL   "Hold Mute/Select\nTo turn off"
+
+#define DISPLAY_STR_L1_QUICK_CALIB  "QuickCalibration"
+#define DISPLAY_STR_L1_FULL_CALIB   "FullCalibration"
+#define DISPLAY_STR_L1_TUBES        "Disconnect tubes"
+#define DISPLAY_STR_L1_REDUCE_O2    "Reduce O2 to 0"
+#define DISPLAY_STR_L1_INCREASE_O2  "incr. O2 to 100%"
+#define DISPLAY_STR_L1_REMOVE_O2    "remove O2 supply"
+#define DISPLAY_STR_L1_CONNECT_N2   "connect N2 100%"
```
Comment (jonasfehr) [inline code/drivers/fpga/display.h:17]: I think this macro/message has become obsolete, but we can keep it for now.

**Review by jonasfehr — APPROVED** [review]
I havn't tested the code as such, but I can't find anyting poping out.

File: code/drivers/fpga/display.c [hunk 2]
```diff
@@ -238,6 +246,14 @@ void display_format_battery_gauge(uint8_t charge_percent)
...
+{
+  make_custom_char(DISPLAY_CHAR_BATTERY_FAULT);
+
+  // Ensure the indicator shows on the display if overwritten
+  s_display[DISP_BATTERY] = (char)CUSTOM_CHAR_BATTERY_INDICATOR;
+}
+
 void display_format_pressure_bar(uint16_t pressure_cmH2O, uint16_t peak_pressure_cmH2O)
 {
   static uint16_t last_pressure = 0;
```
Comment (luisi-at) [inline code/drivers/fpga/display.c]: For consistency, `uint` assignments should include the `u` suffix to integer literals.

File: code/drivers/fpga/display.c [hunk 3]
```diff
@@ -356,6 +372,104 @@ void display_format_pressure_bar(uint16_t pressure_cmH2O, uint16_t peak_pressure
...
+      }
+
+      size_t offset = i;
+
+      if (string_to_display[i] == '\n')
+      {
+        offset += 1u;
+      }
+
+      for (i = 0u; i < (sizeof(buffer) / 2u); i++)
```
Comment (luisi-at) [inline code/drivers/fpga/display.c]: Can this division by 2 be calculated outside of the loop statement? More for clarity than best practice.

Comment (donaldrobson) [reply code/drivers/fpga/display.c]: I've replaced the sizeof with DISP_LEN

**Pull request merged**